src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -43,4 +43,16 @@ public void setData(Model model) {
     protected void indicateAttemptToExecuteIncorrectCommand() {
         EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent(this));
     }
+    
+    /**
+     * Does this command modify data in TaskBook? true or false
+     * @return
+     */
+    public abstract boolean modifiesData();
```

Comment:
> change attribute to an abstract mathod, so when new commands are extended from Command eclipse will remind developer to state whether the command modifies data

One of the big problems with this approach is that is opens up the possibility that a command may declare that it does not modify data, but due to sloppy programming, it actually does. Is there a less intrusive way to do this that does not involve making all Commands implement an extra method?

Hmm, stepping back a bit, we know that whenever a Command modifies model data, it must call the relevant method of the model. e.g., `Model::setEventTask()`, `Model::removeDeadlineTask()` etc. If it does not call any of these methods, then we know for sure that it did not modify any of the model data at all. Perhaps `ModelManager` may be a better place to keep track of whether commands are modifying data or not?